@Test
public void testCloseWithException() {
    try {
        CommonUtil.close(() -> {
            throw new IOException("Test IOException");
        });
        fail("exception expected");
    } catch (IllegalStateException ex) {
        assertEquals("Cannot close the stream", ex.getMessage(), "Invalid exception message");
    }
}
